ls1intum/Artemis
# `Programming exercises`: Fix saving manual assessment after looking at automatic feedback

### Checklist
#### General


#### Client

### Motivation and Context
Fixes #6674.

### Description
The feedback view adds a reference from the submission and the feedbacks to the result to make displaying them easier. Before the result can be sent back to the server as JSON the circular references `result` → `submission`/`feedback` → `result` have to be removed again.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming Exercise with
  - manual assessment enabled
  - at least one student participation

1. Enter assessment view for a submission.
2. Add a comment to some line of code.
3. View results of (automatic) assessment by clicking the link on top right below the buttons.
4. Close the overlay window.
5. Press either Save or Submit at the top right.
6. No error is shown, the assessment has been submitted/saved.

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage






| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|-----------------------------:|
| `result.utils.ts` | 100% changed lines | ✅                           |


### Screenshots
n/a
